pkg/blockbuilder/blockbuilder.go

```diff
@@ -332,6 +338,14 @@ func (b *BlockBuilder) consumePartitionSection(
 	// Thus, truncate the timestamp with ConsumptionInterval here to round the block's range.
 	blockEnd := sectionEndTime.Truncate(b.cfg.ConsumeInterval)
 
+	// If the last commit offset has already reached the offset, that marks the end of the cycle, bail. This can happen when the partition
+	// was scaled down (it's inactive now), but block-builder was lagging, and it chopped this cycle into sections. After a cycle section
```

Comment:
Is it specifically when there was sufficient backlog to make >1 consumption section?

Reply:
nit: I think this whole comment is adding a complex explanation to a simple situation: start==end means nothing to consume!

Reply:
Yes, exactly. Note that, though, the `nextConsumeCycle` checks that the partition's lag is none zero. So the part about having more than one cycle section is important to trigger the bug.